src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -675,21 +675,24 @@ private IDynamicMemberReferenceOperation CreateBoundDynamicMemberAccessOperation
         }
 
         private IDynamicMemberReferenceOperation CreateBoundDynamicMemberAccessOperation(
-            BoundExpression receiver,
+            BoundExpression receiverOpt,
             ImmutableArray<TypeSymbol> typeArgumentsOpt,
             string memberName,
             SyntaxNode syntaxNode,
             ITypeSymbol type,
             ConstantValue value,
             bool isImplicit)
         {
-            Lazy<IOperation> instance = new Lazy<IOperation>(() => Create(receiver));
+            Lazy<IOperation> instance = receiverOpt == null || receiverOpt.Kind == BoundKind.TypeExpression ? 
```

Comment:
Nit: Consider `receiverOpt?.Kind == BoundKind.TypeExpression` as below.

Reply:
> Consider receiverOpt?.Kind == BoundKind.TypeExpression as below.

This is not going to work here because it will yield false when ```receiverOpt == null```, whereas the current condition yeilds true. 

---
In reply to: [151227912](https://github.com/dotnet/roslyn/pull/23170#discussion_r151227912) [](ancestors = 151227912)